src/components/CIPipelineN/CreatePluginModal/utils.tsx

```diff
@@ -0,0 +1,203 @@
+    const requiredFieldValidation = requiredField(version)
+    if (!requiredFieldValidation.isValid) {
+        return requiredFieldValidation
+    }
+
+    return validateSemanticVersioning(version)
+}
+
+export const pluginCreatableTagSelectStyles = {
+    ...commonSelectStyles,
+    valueContainer: (base) => ({
+        ...commonSelectStyles.valueContainer(base),
+        gap: '4px',
+        paddingBlock: '4px',
+    }),
+    control: (base, state) => ({
+        ...commonSelectStyles.control(base, state),
+        minHeight: '36px',
+    }),
+    menuList: (base) => ({
+        ...base,
+        padding: '4px 0px 0px 0px',
+        cursor: 'pointer',
+    }),
+    option: (base, state) => ({
+        ...base,
+        height: '36px',
+        padding: '8px 0px',
+        backgroundColor: state.isFocused ? 'var(--N100)' : 'white',
+        color: 'var(--N900)',
+        cursor: 'pointer',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+
+        ':active': {
+            backgroundColor: 'var(--N100)',
+        },
+    }),
+    multiValue: (base, state) => ({
+        ...base,
+        border: getIsTagValid(state.data.value) ? `1px solid var(--N200)` : `1px solid var(--R500)`,
+        borderRadius: `4px`,
+        background: getIsTagValid(state.data.value) ? 'white' : 'var(--R100)',
+        height: '28px',
+        maxWidth: '250px',
+        margin: 0,
+        paddingLeft: '2px 4px',
+        fontSize: '12px',
+    }),
+}
+
+export const PluginCreatableTagClearIndicator = (props) => (
+    <components.ClearIndicator {...props}>
+        <ICClose className="icon-dim-16 fcn-6 dc__no-shrink" />
+    </components.ClearIndicator>
+)
+
+const getCreatePluginPathArgPortMapping = (
+    inlineStepDetail: InlineStepDetailType,
+): CreatePluginPayloadPipelineScriptDTO['pathArgPortMapping'] => {
+    const portMapList: CreatePluginPayloadPipelineScriptDTO['pathArgPortMapping'] =
+        inlineStepDetail.portMap?.map((portMap) => ({
+            typeOfMapping: PathPortMappingType.PORT,
+            portOnContainer: portMap.portOnContainer,
+            portOnLocal: portMap.portOnLocal,
+        })) || []
+
+    const filePathMapList: CreatePluginPayloadPipelineScriptDTO['pathArgPortMapping'] =
+        inlineStepDetail.mountPathMap?.map((filePathMap) => ({
+            typeOfMapping: PathPortMappingType.FILE_PATH,
+            filePathOnDisk: filePathMap.filePathOnDisk,
+            filePathOnContainer: filePathMap.filePathOnContainer,
+        })) || []
+
+    const dockerArgMapList: CreatePluginPayloadPipelineScriptDTO['pathArgPortMapping'] =
+        inlineStepDetail.commandArgsMap?.map((commandArg) => ({
+            typeOfMapping: PathPortMappingType.DOCKER_ARG,
+            command: commandArg.command,
+            args: commandArg.args,
+        })) || []
+
+    return portMapList.concat(filePathMapList).concat(dockerArgMapList)
+}
+
+export const getCreatePluginPayload = ({
```

Comment:
This method is a cause of concern since it is a separate parser which should actually follow structure of ci-pipeline/patch, but since API is enforcing a separate structure/parser for fields like port mapping, etc, so have to create a separate parser.